app/models/user.rb

```diff
@@ -76,6 +87,14 @@ def email=(value)
     write_attribute :email, (value ? value.downcase : nil)
   end
 
+  def provider=(value)
+    write_attribute :provider, (value ? value.downcase : nil)
+  end
+
+  def uid=(value)
+    write_attribute :uid, (value ? value.downcase : nil)
+  end
+
```

Comment:
Do you really need these methods? Are you getting `provider` and `uid` with different casing from facebook?